internal/objects/gateway/gateway.go

```diff
@@ -78,6 +78,10 @@ func ContourForGateway(ctx context.Context, cli client.Client, gw *gatewayv1alph
 		return nil, nil
 	}
 
+	if gc.Spec.ParametersRef == nil {
```

Comment:
Something to make sure we're ok with. In the past, we've allowed an empty spec and used a set of defaults, but now it's required. 

The problem is knowing if we're going to use GatewayAPI's in contour or not. Today that's defined by specifying the `gateway` bits in the config file.

Reply:
yeah the validation in the operator doesn't let the parametersref be nil, so at this point in the code we just assumed it was set, which causes panics (before this PR this was guarded by the check that the operator had admitted the gw class, which can only happen if there is a parametersref)

i'm thinking this stopgap is ok since we're going to be removing the controllers altogether from the operator soon and in practice if you were using the operator + gw api then you had to make a gw class with parametersref already

Reply:
We should also update the example so pages like this still work: https://projectcontour.io/guides/gateway-api/

Reply:
I think these work currently? gotta try it again but looks like the gwclasses referenced in the examples properly reference a parametersref

Reply:
yeah looks like things are working ok with this change